### ISSUE TITLE
Implement HarvardKey for staff portions of CURIOSity

### ISSUE THREAD

**Issue by VanessaVenti** [harvard-lts/CURIOSity#20]
As a Harvard Library staff member who works on a CURIOSity site, I would like to sign in securely via HarvardKey rather than having to create and remember a separate sign on for CURIOSity.

Expose to Harvard Library staff (not general public requesting XIDs; will roll out to non-HL staff when HarvardKey Light is available.)

Users sign in at: curiosity.lib.harvard.edu/users/sign_in
Users sign out at: curiosity.lib.harvard.edu/users/sign_out

See [subtasks in old Jira story](https://jira.huit.harvard.edu/secure/RapidBoard.jspa?rapidView=218&view=planning&selectedIssue=CURIOSITY-167&issueLimit=100) for what has been done previously.

**Comment by phil-plencner-hl** [comment]
I met with Terry Connolly and Masha Shoykhet in IAM today.
I believe we have everything moving forward at this point. 

I filled out a new CAS application (using the new version of their form) to avoid further confusion. It has been attached to the SNow ticket.

Terry will be creating the Grouper groups. Masha will be setting up the CAS endpoints.

It will be set up to be used in our dev/qa/prod environments as well as Scientist.com's staging environment.

**Comment by phil-plencner-hl** [comment]
Email from Terry: 

Hi Phil
 
The application authorization groups for Harvard Digital Collections/CURIOSity have been set up in the harvard:org:ca:huit:lts:apps:hl-curiosity folder. I am assigning this ticket over to our IAM Engineering team to implements the HarvardKey SSO using the application authorization filter harvard:org:ca:huit:lts:apps:hl-curiosity:authorized-users-hl-curiosity. Please take a look at the "authorized-users-hl-curiosity" group to spot check that the admin users for the application are in that group. 
 
Thank you.
~Terry

**Comment by phil-plencner-hl** [comment]
Hi Terry,
 
I have confirmed that the authorized-users-hl-curiosity has all of CURIOSity’s current administrators as members.
 
Thank you.
 
Phil

**Comment by phil-plencner-hl** [comment]
Dear Phil,


Your request to register your web application(s) with the HarvardKey Authentication System (formerly known as the PIN System) has been processed.  If you need to make any changes to the registration in the future (auto-login timeout, etc.) please provide your application endpoint URL:

https://curiosity-dev.lib.harvard.edu/
https://curiosity-qa.lib.harvard.edu/
https://curiosity.lib.harvard.edu/
[https://staging-harvard-curiosity.notch8.cloud](https://urldefense.proofpoint.com/v2/url?u=https-3A__staging-2Dharvard-2Dcuriosity.notch8.cloud&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=mIByBtxWskN4l9lDOpyT9964APgsOiD0Rsax0fvcA34&m=U6edQpuOD_0GKycB8WwnWHHt210izsaTnenh4BGCkN9fsjJ5izChM7xKHTpPZrzy&s=d6155pkj-XqNY9FhBmO19LIazw4YGlZcgpcAOb8cjgQ&e=)

 

Access to your application through HarvardKey is restricted by the following authorization filter:

harvard:org:ca:huit:lts:apps:hl-curiosity:authorized-users-hl-curiosity

 

When attempting to access pages of your application using the HarvardKey Authentication System, users will be redirected to the HarvardKey login page.  What is “secured” depends on your configuration in your CAS client.  After successful authentication at the login page, the user will be redirected back to your application.

Additional developer documentation for the HarvardKey Authentication System is available at: http://iam.harvard.edu/get-started/authentication

Other useful URLs:

CAS Server Login URL: [https://www.pin1.harvard.edu/cas/login](https://ww.pin1.harvard.edu/cas/login)
CAS Server URL Prefix for Service Ticket (ST) Validation: https://www.pin1.harvard.edu/cas
CAS Logout URL: [https://www.pin1.harvard.edu/cas/logout](https://stage.pin1.harvard.edu/cas/logout)

In order to protect specific prefixes or paths, you will need to configure the CAS client within your environment.

As always, please do not hesitate to be in touch if you have any questions.

Sincerely,



Alexander Nazarenko
Harvard University Information Technology
HUIT Support (617) 495-7777
http://huit.harvard.edu/

**Comment by dl-maura** [comment]
@orangewolf We have this up on dev!

It does work (I am getting HKey and signed in) but some weirdness:

- I was an admin automatically. I had purged the roles and users table, so not sure why, do I need to purge other things?
- The manage users page is broken (see screenshot)
![Screen Shot 2022-08-04 at 4 39 42 PM](https://user-images.githubusercontent.com/61880785/182956980-1185d8bc-bfb9-4eac-bf51-4e7124b2420f.png)

**Comment by dl-maura** [comment]
Above issue was fixed but found a secondary issue:

Attempted to make a curator of a collection and got this error
![Screen Shot 2022-08-05 at 9 25 13 AM](https://user-images.githubusercontent.com/61880785/183086791-a22287ff-4e5b-420d-a8ac-0d1c22978759.png)

**Comment by dl-maura** [comment]
Noted in slack as well:
Possibly just a visual bug? VV is an admin but I can still choose to make her an admin. Only her account has this, the others don't. And she is able to do all admin things:


![Image](https://user-images.githubusercontent.com/61880785/183487469-1095e78c-88bc-4645-904f-1e95faa1c69c.png)

**Comment by VanessaVenti** [comment]
@phil-plencner-hl and @dl-maura - does this all look good on your end? If so, you can close this one.

**Comment by phil-plencner-hl** [comment]
@VanessaVenti @dl-maura I updated the check-boxes in @enriquediaz's earily comment (https://github.com/harvard-lts/CURIOSity/issues/20#issuecomment-1178074814) to be complete (with the exception of "CURIOSity unauthenticated users page styling". I'm not sure what this entails and if it is completed?

I also added a new check-box for "Onboard HarvardKey for CURIOSity BETA" as I realized today when I had IAM set up HarvardKey I did not include the beta url (http://beta.curiosity.lib.harvard.edu/) as an endpoint.

I sent an email to Emily Kelly today to see how we can set that up.

I'm not sure if we should close this and make a separate task for confirming live/beta HarvardKey is working as expected or if we should leave this one open? I don't really have a strong preference.
